enhancements/cluster-logging/fine-grained-per-namespace-logs-access.md

```diff
@@ -0,0 +1,249 @@
+
+## Proposal
+
+Use the `ClusterLogging` custom resource definition managed by the Cluster Logging Operator to enable/disable the fine grained access to logs for the LokiStack logstore. This will be achieved by adding a new field to the CRD called `advancedLogsAccess`. This new field is a boolean with a default value of false.
+
+* For the first use case, where we have non-admin users, the proposed solution is to create a group for application log readers called `log-reader-group`, and grant this group the ability to read application logs by binding it to the existing ClusterRole `logging-application-logs-reader`. The cluster admin can then either add a user to the `log-reader-group` thus granting access to application logs on all namespace where they have access, or create a role binding to the user on each namespace they want to grant access to logs on.
+
+* For the second use case, where we have users with admin like privileges, the proposed solution is to create a new group called `restricted-cluster-admin-group`. This group has all the permissions a cluster admin has, except for application logs. We can then grant this group access to application logs on specific namespace by binding it to `logging-application-logs-reader` on the desired namespace. 
+
+### Workflow Description
+
+1. The cluster administrator enables fine grained logs access in the `ClusterLogging` resource:
+
+```yaml
+apiVersion: logging.openshift.io/v1
+kind: ClusterLogging
+metadata:
+  name: instance
+  namespace: openshift-logging
+spec:
+  logStore:
+    type: lokistack
+    lokistack:
+      name: lokistack-dev
+      advancedLogsAccess: true
+  ...
+```
+
+2. The Cluster Logging Operator (CLO) deletes the `logging-all-authenticated-application-logs-reader` ClusterRoleBinding. This binding allows all authenticated users to see application logs in namespace where they have access.
+
+3. The CLO then creates a new empty group `log-reader-group` and binds it to `logging-application-logs-reader`:
+
+```yaml
+apiVersion: user.openshift.io/v1
+kind: Group
+metadata:
+  name: log-reader-group
+users:
+```
+The group contains no users. The cluster admin can then add users to the group by editing it manually.
+
+```yaml
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: logging-application-logs-reader-binding
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: logging-application-logs-reader
+subjects:
+- kind: Group
+  name: log-reader-group
+  apiGroup: rbac.authorization.k8s.io
+```
+
+4. The operator also creates a new ClusterRole called `restricted-cluster-admin` and binds it to a new group `restricted-cluster-admin-group`. `restricted-cluster-admin` grants access to all api objects except for logs:
+
+```yaml
+kind: ClusterRole
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: restricted-cluster-admin
+rules:
+- apiGroups:
+  - ''
+  - 'admissionregistration.k8s.io/v1'
```

Comment:
What is the significance of these groups and how do they relate to viewing admin logs?  This looks like we have redefined an admin role that is a subset of something else.  Again, consider the use of virtual resources as I am [proposing](https://github.com/openshift/enhancements/pull/1410/files#diff-9d07913b066ab6b25695e5fcd8e023e05ef0761832052d53cc76cf826f699448R81-R91).  

* The operator can check if a user or group satisfies that role
* An administrator can create a role binding to any user or group
* Conceptionally we can apply the same logic to namespaced roles and bindings to restrict namespaces